pytest/tests/loadtest/locust/common/base.py

```diff
@@ -115,8 +130,100 @@ def sign_and_serialize(self, block_hash) -> bytes:
+            "method": "broadcast_tx_async",
+            "params": [base64.b64encode(signed_tx).decode('utf8')],
+            "id": "dontcare",
+            "jsonrpc": "2.0"
+        }
+
+        meta = {
+            "request_type": "near-rpc",
+            "name": locust_name,
+            "start_time": time.time(),
+            "response_time": 0,  # overwritten later  with end-to-end time
+            "response_length": 0,  # overwritten later
+            "response": None,  # overwritten later
+            "context": {},  # not used  right now
+            "exception": None,  # maybe overwritten later
+        }
+        start_perf_counter = time.perf_counter()
+
+        # async submit
+        submit_raw_response = requests.post(url="http://%s:%s" %
+                                            self.node.rpc_addr(),
+                                            json=j)
+        meta["response_length"] = len(submit_raw_response.text)
+
+        # extract transaction ID from response, it should be "{ "result": "id...." }"
+        submit_response = submit_raw_response.json()
+        if not "result" in submit_response:
+            meta["exception"] = RpcError(submit_response,
+                                         message="Didn't get a TX ID")
+            meta["response"] = submit_response.content
+        else:
+            tx.transaction_id = submit_response["result"]
+            try:
+                # poll for tx result, using "EXPERIMENTAL_tx_status" which waits for
+                # all receipts to finish rather than just the first one, as "tx" would do
+                j = {
+                    "method": "EXPERIMENTAL_tx_status",
+                    "params": [tx.transaction_id,
+                               tx.sender_id()],
+                    "id": "dontcare",
+                    "jsonrpc": "2.0"
+                }
+                # using retrying lib here to poll until a response is ready
+                self.poll_tx_result(meta, j)
+            except NearError as err:
+                logging.warn(f"marking an error {err.message}, {err.details}")
+                meta["exception"] = err
+
+        meta["response_time"] = (time.perf_counter() -
+                                 start_perf_counter) * 1000
+
+        # Track request + response in Locust
+        self.request_event.fire(**meta)
+        return meta["response"]
+
+    @retry(wait_fixed=500,
+           stop_max_delay=DEFAULT_TRANSACTION_TTL_MILLISECONDS,
+           retry_on_exception=is_tx_unknown_error)
+    def poll_tx_result(self, meta, json):
+        result_response = requests.post(url="http://%s:%s" %
+                                        self.node.rpc_addr(),
+                                        json=json)
+        # set raw response here in case an error is raised below
+        meta["response"] = result_response.content
+        # this may raise a `NearError`
+        meta["response"] = evaluate_rpc_result(result_response.json())
```

Comment:
I agree that this does the job, but it is a bit tricky to read and verify that this code is correct.
An alternative way would be to re-raise the exception:
```python
try:
  meta["response"] = evaluate_rpc_result(result_response.json())
except:
  # Store raw response to improve error-reporting.
  meta["response"] = result_response.content
  raise
```